File: solutions/python/04/floyd_warshall/main.py
def floyd(dist):
    n = len(dist)
    for k in range(n):
        for i in range(n):
            for j in range(n):
                if dist[i][k] < 10**17 and dist[k][j] < 10**17:
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
    return dist


def main() -> None:
    INF = 10**17
    dist = [
        [0, 3, INF],
        [INF, 0, 1],
        [2, INF, 0],
    ]
    print(floyd(dist))


if __name__ == "__main__":
    main()

File: solutions/go/04/floyd_warshall/main.go
package main

import "fmt"

func main() {
	const INF = 1 << 60
	dist := [][]int{
		{0, 3, INF},
		{INF, 0, 1},
		{2, INF, 0},
	}
	n := len(dist)
	for k := 0; k < n; k++ {
		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				if dist[i][k] >= INF || dist[k][j] >= INF {
					continue
				}
				if dist[i][j] > dist[i][k]+dist[k][j] {
					dist[i][j] = dist[i][k] + dist[k][j]
				}
			}
		}
	}
	fmt.Println(dist)
}
